GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -240,55 +240,68 @@ public bool Apply(RevisionFilter filter)
                 Debug.Fail("Not supported");
             }
 
+            var searchParametersChanged = filter.FilterByAuthor != ByAuthor
+                                          || filter.FilterByCommitter != ByCommitter
+                                          || filter.FilterByCommit != ByMessage
+                                          || filter.FilterByDiffContent != ByDiffContent;
+
             if (filter.FilterByAuthor)
             {
-                if (ByAuthor && string.Equals(Author, filter.Text, StringComparison.CurrentCulture))
```

Comment:
As it is in master, ByAuthor and similar must be set for Author to be set. If this check is not needed, the explicit setting should be removed. Maybe `public bool ByAuthor { get; private set; }`
FilterByAuthor is related to disbling/enabling options without clearing the contents.

Reply:
Sorry, after reading multiple times carefully, I don't get it.

Reply:
> As it is in master, ByAuthor and similar must be set for Author to be set. If this check is not needed, the explicit setting should be removed. Maybe `public bool ByAuthor { get; private set; }`

In fact, I explicitly take care of setting `ByAuthor` before setting the value (and setting to `false` when it has been unselected).

That's because this test is used for 2 different things ( returning if the filter should be applied and bypassing the set of properties in the following code) that should not behave exactly the same way that cause the bad behavior.
If you return because you are in one of the if, you never update the later properties and do not launch the search (early return `false` even if in fact, you should)

Reply:
Works as it should - restores 3.5 behavior, changed below though.